aten/src/ATen/native/transformers/cuda/sdp_utils.cpp

```diff
@@ -561,11 +561,21 @@ bool can_use_cudnn_attention(const sdp_params& params, bool debug) {
   return true;
 }
 
-bool can_use_flash_attention(sdp_params const& params, bool debug) {
-#ifndef USE_FLASH_ATTENTION
-  TORCH_WARN_ONCE(!debug, "Torch was not compiled with flash attention.");
+bool is_flash_attention_available() {
+#ifdef USE_FLASH_ATTENTION
+  return true;
+#else
   return false;
 #endif
+}
+
+bool can_use_flash_attention(sdp_params const& params, bool debug) {
+  if (!is_flash_attention_available()) {
```

Comment:
while we are at it, I actually think that there is a existing bug here. I think this `is is_flash_attention_available` shortcut return should only be ran if "if params.query.device == cuda"

Reply:
Not sure I understand. There is already the `check_all_tensors_on_device` constraint which causes this function to return false if `params.query.device` isn't a CUDA tensor. I don't see how the change you propose would change the behavior of this function, except for which message is printed in some instances. Is that your goal?

Reply:
sorry again for some reason I read this before as mutating: https://github.com/pytorch/pytorch/blob/e3dc20c94bbbeb6aa4782446c9c7115896002adb/torch/csrc/Module.cpp#L1988

which has the bug I mentioned .

This PR looks good